### DISCUSSION TITLE
Remove TODO, FIXME, etc., ligatures

### DISCUSSION THREAD

**Discussion by nomasprime** [fabrizioschiavi/pragmatapro#194] [General]
These create too many edge cases and inconstant output.

They're a user and, I'd imagine, maintainer minefield.

Personally I've stopped using the ligature variant of the font because of the inconsistent and undesirable output. 

Example in Ansible.

This renders a ligature:

```
[WARNING]: Platform darwin on host localhost is using the discovered Python interpreter at /usr/bin/python, but future
installation of another Python interpreter could change this. See
https://docs.ansible.com/ansible/2.9/reference_appendices/interpreter_discovery.html for more information.
```

This does not:

```
[DEPRECATION WARNING]: Invoking "apt" only once while using a loop via squash_actions is deprecated. Instead of using a
 loop to supply multiple items and specifying `name: "{{ item }}"`, please use `name: ['apt-transport-https', 'ca-
certificates', 'curl', 'software-properties-common']` and remove the loop. This feature will be removed in version
2.11. Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
```

Or we could continue making requests and patching the font for every single variant but seems kinda pointless.

**Comment by mariojackson** [comment]
I understand your pain point, but for the working cases, I would not like for them to be removed. It makes it much easier for me to find the messages in a log, espescially if there are no linebreaks. Just my opinion though, but I really like these ligatures.

**Reply by nomasprime** [reply]
Personally find consistency easier.

I liked the idea initially but ultimately more cons than pros. Stopped using ligatures altogether and terminal's much snappier, especially when combined with GPU rendering.

**Comment by wismill** [comment]
I agree with @mariojackson. 
@nomasprime how many edge cases have you found? It may be worth to add them to the font if they are not too numerous.

**Reply by nomasprime** [reply]
Sorry, forgot about this issue.

Can't remember exactly but too many and ligatures in general were negatively impacting my workflow, stopped using them altogether.

**Reply by fabrizioschiavi** [reply]
I'm working at the next release. I'll remove these ligatures as default. 
It's a pain but hestetical solutions can't be obstacles

**Reply by nomasprime** [reply]
Thanks @fabrizioschiavi, agree. Sure users that need ligatures will appreciate having the option.

Looking forward to the next release.

**Comment by michaelcadilhac** [comment]
One I often see is with my tests, where I have [FAIL] but also [PASS].  The latter is not included.  I started using [KO] [OK] to avoid the discrepancy, tho I'd love to see these two as ligatures too.

**Comment by mrclayman** [comment]
Personally, I don't care much about ligatures in general, but they are nice to have and I do like those that mark log levels.

@michaelcadilhac , feel free to shoot Fabrizio a request on the [PragmataPro page](https://www.fsd.it/shop/fonts/pragmatapro/). I requested the [FAIL] ligature and it got included. I am sure Fabrizio will be able to work [PASS] in as well, although it's hard to say what the ETA of a new version of the font will be. :+1:

**Reply by nomasprime** [reply]
This was my conclusion on ligatures:

- Inconsistencies actually make scanning harder, therefore slower
- Terminals are snappier without ligature rendering, especially with GPU rendering enabled
- Modal editing with motions is slower when you have to make the extra mental step of mapping the actual number of chars to the rendered ligature

Still use the font but I don't use ligatures anymore.
